src/Engine/ProtoCore/DSASM/InstructionSet.cs

```diff
@@ -110,6 +110,111 @@ public struct MetaData
         public int type;
     }
 
+    /// <summary>
+    /// Wrapper over all values handled by the MSIL/CLR engine.
+    /// </summary>
+    internal struct CLRStackValue
```

Comment:
Simplified version of existing StackValue structure in the old VM.
Reasons for creating/using this new struct:
1. Simplifies the copy/paste process from old VM logic to new one.
2. Replication needs access to particular information that we can store once rather than have it compute every time (ex rank)

Reply:
maybe related to:
https://jira.autodesk.com/browse/DYN-5152

@aparajit-pratap noted yesterday that these types are still not returned from `ReplicationLogic` but only used internally to compute function endpoints / replication behavior - is that still the case?

Reply:
That is still the case.
But this does incur some performance penalties due to the extra Marshaling/Unmarshaling that has to happen for every returned value and input argument